### PR TITLE
Crafting station: merge conflict edition

### PR DESCRIPTION
**Pull request by Omni44**
## About The Pull Request

Adds crafting station machine which allows user to craft junk ammo in the same fashion junk ammo kit does, but with more favourable formula (~22% better even with basic parts), but also gun and armor parts. Machine requires 1 micro laser, 1 micro manipulator and single matter bin, and its efficiency, work time and storage capacity scale with quality of parts used, and also scales with gunsmith perk. Circuit board is available on technomancer cirtuits disk and should be findable in maint, but I am not 100% sure about that.
3dprinter

Thanks to Baron for sprites, SPCR, Shown, Alli and Frost for tardwrangling me with this.
## Why It's Good For The Game
There's no efficient way to craft gunparts as is, which is very bad thing, as is they are harder/more expensive to come across than actual guns, which with direction of the development leaning more towards prolfieration of makeshfit guns is an issue that direly needed fixing for a long while now. Plus I hope it will encourage people to make their own bases/sweatshops more.

LINK to previous version and related discussion: #6718

## Changelog
:cl:
add: Added crafting station&appropraite circuit board
tweak: tweaked some gun crafting recipes a bit
/:cl:

### CODE REVIEW

**Comment by Omni44** [comment]
Review ready now, I think. I added deconstruction, part replacement and scaling of properties with quality of parts, it should behave the way machine should. Only missing detail is it does not have unscrewed panel sprite, I forgot about that completely.

File: code/game/machinery/crafting_station.dm [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+	options[".25 Caseless Rifle ammunition"] = "clrifle"
+	options[".30 Rifle ammunition"] = "lrifle"
+	options[".35 Auto ammunition"] = "pistol"
+	options[".40 Magnum ammunition"] = "magnum"
+	options[".50 Shotgun Buckshot ammunition"] = "shot"
+	options[".50 Shotgun Beanbag ammunition"] = "bean"
+	options[".50 Shotgun Slug ammunition"] = "slug"
+	options[".60 Anti-Material ammunition"] = "antim"
+	options["Gun parts"] = "gunpart"
+	options["Armor parts"] = "armorpart"
```
Comment (TheShown911) [inline code/game/machinery/crafting_station.dm:84]: Can you make an associative list by just having 
```
	var/list/options = list(
	".20 Rifle ammunition" = "srifle",
	".25 Caseless Rifle ammunition" = "clrifle",
	".30 Rifle ammunition" = "lrifle",
	".35 Auto ammunition" = "pistol",
	".40 Magnum ammunition" = "magnum",
	".50 Shotgun Buckshot ammunition" = "shot",
	".50 Shotgun Beanbag ammunition" = "bean",
	".50 Shotgun Slug ammunition" = "slug",
	".60 Anti-Material ammunition" = "antim",
	"Gun parts" = "gunpart",
	"Armor parts"= "armorpart",
	)
```
Or something, this isn't your problem as many other associative lists in the code do this, but it bugs me.

File: code/game/machinery/crafting_station.dm [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+	if(choice == null)
+		return
+
+	var/produce_type = options[choice]
+
+	switch(produce_type)
+		if("pistol")
+			for(var/material in needed_material_ammo)
+				var/amount = needed_material_ammo[material]
+				if(amount>stored_material[material])
```
Comment (TheShown911) [inline code/game/machinery/crafting_station.dm:96]: Not putting space between operators is a crime against humanity
`if(amount > stored_material[material])`

File: code/game/machinery/crafting_station.dm [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+			boxxes = 1
+			mags = 1
+		if(42 to 48)
+			boxxes = 1
+			piles = 2
+			mags = 1
+		else
+			boxxes = 2 + round(dice/10-5,1)
+
+	if(rifle==1) //srifle
```
Comment (TheShown911) [inline code/game/machinery/crafting_station.dm:352]: `if(rifle == 1)`
And everywhere else too.

File: code/game/machinery/crafting_station.dm [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
+		las_rating += M.rating
+
+	work_time = initial(work_time) - 2*(man_rating - 1) SECONDS
+	mat_efficiency = initial(mat_efficiency)-(1.5^las_rating)
+
+/obj/machinery/autolathe/on_deconstruction()
+	for(var/mat in stored_material)
+		eject(mat, stored_material[mat])
+..()
```
Comment (TheShown911) [inline code/game/machinery/crafting_station.dm:572]: Linter failure, probably bad indentation.

**Review by TheShown911 — COMMENTED** [review]
Fix PR description with cool gifs and stuff so rosa can exploit it for the devlog. Code is mostly fine, just nitpicks.

**Review by drexample — APPROVED** [review]

**Pull request merged**